### PR TITLE
Add various limits to GNTDeposits

### LINKED ISSUE #40: Implement limits in Golem Deposit Contract
We'd like to introduce flexible limits to the Concent's GNT Depost Contract

* Add a private variable, `maximum_deposits_total`, to the contract to limit the maximum cumulative amount of GNTB under Contract's control anytime.
** when a call is made to register a new deposit or add to an existing deposit, this value should be consulted and in case the total amount of deposits exceeds this amount, the deposit should be disallowed,
** the default value should be null, in which case, the contract has no maximum limit,
** there a setter this value, available only to the owner of the contract
* Secondly, add a private variable, `maximum_deposit_amount` to the contract to limit the value of a single deposit
** when a call is made to register a new deposit or add to an existing deposit, this value should be consulted and in case the deposit would exceed the amount, the deposit should be disallowed,
** the default value should be null, in which case, the contract has no limit on a single deposit
** there a setter this value, available only to the owner of the contract

* there should be a call to verify if a requested deposit will be able to be registered

* Add a daily payout limit for reimbursements made by Concent from the deposit over the period of 24 hours, working in an analogous way as the above two values.

### PR DESCRIPTION
**Pull request by Krigpl**
Fixes #40

### CODE REVIEW

File: contracts/GNTDeposit.sol [hunk 1]
```diff
@@ -184,11 +218,35 @@ contract GNTDeposit is ReceivingContract, Ownable {
 
     function _reimburse(address _from, address _to, uint256 _amount) private {
         require(balances[_from] >= _amount);
+        if (daily_reimbursement_limit != 0) {
+            if (current_reimbursement_day != block.timestamp / 1 days) {
```
Comment (shadeofblue) [inline contracts/GNTDeposit.sol]: unsure if solidity compiler optimizes that - but wouldn't it be better to calculate the value once? and only compare/assign it later? question 2 -> are those fixed point divisions? what's the type of the right-hand side value? also unit256?

Comment (Krigpl) [reply contracts/GNTDeposit.sol]: No, no, no, uint256, yes

Comment (shadeofblue) [reply contracts/GNTDeposit.sol]: btw, another question - is there a chance for a race condition that could change the value of any compared values (both `block.timestamp` and the limits) while this block is executed?

Comment (shadeofblue) [reply contracts/GNTDeposit.sol]: plus ... if those are not fixed point divisions - is it safe to check for equality?

**Pull request merged**